src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+    });
+  });
+}
+
+MatrixTwitter.prototype._get_user = function (data) {
+  return new Promise((resolve, reject) => {
+    this.app_twitter.get('users/show', data, (error, user) => {
+      if (error) {
+        error = error[0];
+        log.error(
+              'Twitter',
+              "get_profile_by_id: GET /users/show returned: %s %s",
+              error.code,
+              error.message
+            );
+        reject(error.message);
+        return;
+      }
+      this._update_user_timeline_profile(user);
+      resolve(user);
+    });
+  });
+}
+
+/**
+ * MatrixTwitter.prototype.get_profile_by_id - Get a Twitter profile by a users
+ * Twitter ID.
+ *
+ * @param  {number} id Twitter Id
+ * @return {Promise<TwitterProfile>} A promise containing the Twitter profile
+ * to be returned. See https://dev.twitter.com/rest/reference/get/users/show
+ */
+MatrixTwitter.prototype.get_profile_by_id = function (id) {
+  log.info("Twitter", "Looking up T" + id);
+  return this.storage.get_profile_by_id(id).then((profile)=>{
+    if(profile != null) {
+      return profile;
+    }
+    return this._get_user({user_id: id});
+  });
+}
+
+/**
+ * MatrixTwitter.prototype.get_user_by_screenname - Get a Twitter profile by a
+ * users screen name.
+ * @param  {number} id Twitter Screen name
+ * @return {Promise<TwitterProfile>} A promise containing the Twitter profile
+ * to be returned. See {@link}
+ *
+ * @see {@link https://dev.twitter.com/rest/reference/get/users/show}
+ */
+MatrixTwitter.prototype.get_user_by_screenname = function (name) {
+  log.info("Twitter", "Looking up @" + name);
+  return this.storage.get_profile_by_name(name).then((profile)=>{
+    if(profile != null) {
+      return profile;
+    }
+    return this._get_user({screen_name: name});
+  });
+}
+
+/**
+ * MatrixTwitter.prototype.tweet_to_matrix_content - This function will fill
+ * the content structure for a new matrix message for a given tweet.
+ *
+ * @param  {TwitterTweet} tweet The tweet object from the Twitter API See {@link}
+ * @param  {type} type  The 'msgtype' of a message.
+ * @return {object}     The content of a Matrix 'm.room.message' event.
+ *
+ * @see {@link https://dev.twitter.com/overview/api/tweets}
+ */
+MatrixTwitter.prototype.tweet_to_matrix_content = function (tweet, type) {
+  return {
+    "body": new HTMLDecoder().decode(tweet.text),
+    "created_at": tweet.created_at,
+    "likes": tweet.favorite_count,
+    "reblogs": tweet.retweet_count,
+    "tweet_id": tweet.id_str,
+    "tags": tweet.entities.hashtags,
+    "msgtype": type
+  }
+}
+
+//Runs every 500ms to help not overflow the room.
+MatrixTwitter.prototype._process_head_of_msg_queue = function () {
+  if(this.msg_queue.length > 0) {
+    var msgs = this.msg_queue.pop();
+    //log.info("Twitter","Pulling off queue:",msg.content.body);
+    for(var msg of msgs) {
+      var intent = this._bridge.getIntent(msg.userId);
+      intent.sendEvent(msg.roomId, msg.type, msg.content).catch(reason =>{
```

Comment:
This races: there is no guarantee the first `msg` will be sent first. You should probably promise chain this to make sure you do it in a consistent order.